statsmodels/stats/libqsturng/qsturng_.py

```diff
@@ -474,13 +475,18 @@ def _func(a, p, r, v):
     if r == 3:
         f += -0.002 / (1. + 12. * _phi(p)**2)
 
+        if np.isinf(v):
+            # Clip infinite v values
+            v = 1e38
+
         if v <= 4.364:
-            f += 1./517. - 1./(312.*(v,1e38)[np.isinf(v)])
+            f += 1./517. - 1./(312.*v)
         else:
-            f += 1./(191.*(v,1e38)[np.isinf(v)])
```

Comment:
that's not the correct fix
I think we need to convert numpy bool to python bool which should work for indexing a tuple
(or astype(int) should also work)

Reply:
correction: the current fix is correct just not literally the same
still LGTM

Reply:
Glad we're on the same page.  I'll ping on de-facto green.

Couple others similar to this coming up.

Reply:
The method used in the original code is an abomination to decent code.

There are lots more places to fix this bug in this library.  I think I got them all in mine

Reply:
> The method used in the original code is an abomination to decent code.

Seriously.

> There are lots more places to fix this bug in this library. I think I got them all in mine

Yah.  Since I just re-pushed, yours will finish CI first, so I'll rebase and just use your version for this file.

Reply:
> The method used in the original code is an abomination to decent code.

Nope, it's standard advertised python, that numpy doesn't want to follow anymore.
Python's bool behaves like an int, and indexing with bool was a "cute" trick before the conditional assignment was added to Python.

(https://github.com/rogerlew/qsturng-py doesn't seem to be maintained anymore, so I guess we maintain our code without foreign compatibility requirements.)

Reply:
> Nope, it's standard advertised python,

Really not worth arguing about, but consider:
- it was not immediately obvious to you that the version I implemented was functionally equivalent
- one person whose opinion you respect and one person who has two thumbs and strong opinions about the Star Wars Expanded Universe had a visceral reaction to this pattern.
- cute tricks and good code are overlapping but far-from-identical sets